giskard/testing/tests/miscellaneous.py

```diff
@@ -0,0 +1,234 @@
+                sample_rows_copy,
+                target=dataset.target,
+                cat_columns=dataset.cat_columns,
+                column_types=dataset.column_types,
+                validation=False,
+            )
+        )
+
+    # Predictions
+    predictions = []
+    for current_dataset in augmented_df:
+        predictions.append(model.predict(current_dataset).raw)
+    predictions = np.array(predictions)
+
+    # If classification, select the right index in the last dimension
+    if predictions.ndim == 3:
+        predictions = predictions[:, :, classif_index_label]
+
+    return predictions, sample_rows
+
+
+@test(name="Monotonicity", tags=["monotonicity"])
+def test_monotonicity(
+    model: BaseModel,
+    dataset: Dataset,
+    column_names: List[str],
+    column_values: Optional[dict] = None,
+    increasing: bool = True,
+    random_state: int = 123456,
+    num_samples: int = 100,
+    num_grid: int = 50,
+    classif_index_label: int = 0,
+):
+    """Test if the model is monotonic for a given column name by selecting random samples from
+    the dataset and augmenting them with different values for the specified column
+
+    Parameters
+    ----------
+    model : BaseModel
+        Model used to compute the test
+    dataset : Dataset
+        Actual dataset used to compute the test
+    column_names : List[str]
+        Column names to test (at the same time)
+    column_values : Optional[dict]
+        Values to test for each column, the values should always have the same length
+    increasing : bool
+        Whether to test for increasing or decreasing monotonicity
+    random_state : int
+        Random state for sampling
+    num_samples : int
+        Number of samples from which generate the grid and test
+    num_grid : int
+        Number of points in the grid of values for the colum
+    classif_index_label : int
+        If classification, which index to consider for the test
+
+    Returns
+    -------
+    TestResult
+        The test result.
+    """
+
+    _check_columns(column_names=column_names, column_values=column_values)
+
+    # Get predictions
+    predictions, sample_rows = _get_predictions(
+        model=model,
+        dataset=dataset,
+        column_names=column_names,
+        column_values=column_values,
+        random_state=random_state,
+        num_samples=num_samples,
+        num_grid=num_grid,
+        classif_index_label=classif_index_label,
+    )
+
+    # Check for monotonicity
+    if increasing:
+        passed = np.diff(predictions, axis=0) >= 0
+    else:
+        passed = np.diff(predictions, axis=0) <= 0
+
+    # --- debug ---
+    output_ds = list()
+    if not passed.all():
+        output_ds.append(dataset.slice(row_failing_slicing_fn(index_failure=sample_rows.index[~passed.all(axis=0)])))
+    # ---
+
+    return TestResult(passed=passed.all(), output_ds=output_ds, metric=(~passed).mean())
+
+
+@test(name="Smoothness", tags=["smoothness"])
+def test_smoothness(
```

Comment:
for this test, you have to check if the `feature_names` given are numeric (and not text or categorical for instance) by comparing it against `dataset.column_types`